lib/systems/src/MainChargeSystem.cpp

```diff
@@ -3,19 +3,26 @@
 #include <algorithm>
 #include <cmath>
 
-MainChargeSystem::MainChargeSystem(float target_volt, float max_charge_current, float max_cell_temp) : target_voltage(target_volt), max_ac_charging_current(max_charge_current), max_allowable_cell_temperature(max_cell_temp){}
+/**
+ * @brief void returning function that sets current_setpoint in the output struct to a calculated current value based on current cell voltage and temp 
+ * @param ACUAllData_s struct populated with raw data from ACU
+ * @param CCUParams struct with params that control the charge control unit
+ */
+float MainChargeSystem::calculate_charge_current(ACUAllData_s inputValues, CCUParams &chargeParams)
+ {
 
-void MainChargeSystem::charge_cycle(CCUInput_s inputValues, CCUOutput_s &outputValues){
-    if (inputValues.max_cell_voltage >= target_voltage || inputValues.max_cell_temperature > max_allowable_cell_temperature) { //checks whether or not any of the cells are charged yet
-      outputValues.current_flow = false; //this will be function that actually turns on/off current flow 
-      outputValues.current_setpoint = 0;
+    double max_cell_voltage = *std::max_element(inputValues.voltages, inputValues.voltages + (sizeof(inputValues.voltages) / sizeof(inputValues.voltages[0]))); //sizeof is needed because max value takes in a start and end memory location of the array
+    double max_cell_temperature = *std::max_element(inputValues.cell_temperatures, inputValues.cell_temperatures + (sizeof(inputValues.cell_temperatures) / sizeof(inputValues.cell_temperatures[0])));
+
+    if (max_cell_voltage >= chargeParams.target_voltage || max_cell_temperature > chargeParams.max_allowable_cell_temperature) { //checks whether or not any of the cells are charged yet
+      return 0;
     }
     else { 
-      float temperature_scalar = ((max_allowable_cell_temperature - inputValues.max_cell_temperature)/10); //This needs to be replaces with a log function  
-      float distance_from_ideal_voltage = target_voltage - inputValues.max_cell_voltage;
+      float temperature_scalar = ((chargeParams.max_allowable_cell_temperature - max_cell_temperature)/10); //divide by 10 to ensuse that scalar is within 0-1. 
+      float distance_from_ideal_voltage = chargeParams.max_allowable_cell_voltage - max_cell_voltage;
       float voltage_scalar = ((log(distance_from_ideal_voltage+.5))/log(11))+.3;
-      float final_scalar = std::min(temperature_scalar, voltage_scalar);
-      outputValues.current_flow = true; //needs to be replaced with the actual thing to set current flow to the correct value
-      outputValues.current_setpoint = max_ac_charging_current * final_scalar; //currently the PID returns a value greater than 1   
+      float final_scalar = std::min(temperature_scalar, voltage_scalar);  
+      float calculated_current = 25 * final_scalar; //25 is a tentative amp value based on 6kw at 240 volts, may need to be asjusted depending on voltage
```

Comment:
nit: "adjusted" misspelt

Reply:
note: lots of magic numbers in this file. once this gets merged into main, the linter is going to have a field day.